nixos/modules/tasks/network-interfaces.nix

```diff
@@ -612,6 +657,93 @@ in
       };
     };
 
+    networking.wlanInterfaces = mkOption {
+      default = { };
+      example = {
+        wlan-adhoc0 = {
+          base = "wlp6s0";
+          type = "ibss";
+          mac = "02:00:00:00:00:01";
+        };
+      };
+      description =
+        ''
+          This option allows you to define vlan devices that tag packets
+          on top of a physical interface. The value of this option is an
+          attribute set. Each attribute specifies a vlan, with the name
+          specifying the name of the vlan interface.
```

Comment:
Yeah, I guess "virtual" is not necessary.

Related: this option should probably be moved into `networking.wireless.*`.  Note that there already is an option named `networking.wireless.interfaces`.